src/js/components/agencyV2/AgencyOverview.jsx

```diff
@@ -0,0 +1,129 @@
+    const {
+        name,
+        logo,
+        website,
+        mission,
+        congressionalJustification,
+        showAboutData,
+        subtierCount
+    } = useSelector((state) => state.agencyV2.overview);
+
+    const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+
+    useEffect(() => {
+        const handleResize = throttle(() => {
+            setWindowWidth(window.innerWidth);
+        });
+        window.addEventListener('resize', handleResize);
+        return () => window.removeEventListener('resize', handleResize);
+    }, []);
+
+    const image = logo ? (
+        <img
+            className="agency-overview__image"
+            src={`graphics/agency/${logo}`}
+            alt={`${name} logo`} />
+    ) : '';
+
+    const missionBlock = (
+        <div className="agency-overview__data">
+            <h4>Agency Mission</h4>
+            <p>{mission || '--'}</p>
+        </div>
+    );
+
+    const aboutBlock = (
+        <div className="agency-overview__data">
+            <h4>About this Agency&apos;s Data</h4>
+            <p>
+                There is a 90 day delay in displaying contract award data, subcontract data,
+                and Account Breakdown by Award (File C) data for the Department of Defense (DOD).
+                For more information, visit our <Link to="/about?section=data-quality">About Page</Link>.
+                To see a complete list of this agency&apos;s submissions, visit our&nbsp;
+                <Link to="/submission-statistics/agency/097">Submission Statistics page</Link>.
+            </p>
+        </div>
+    );
+
+    const websiteBlock = (
+        <div className="agency-overview__data">
+            <h4>Website</h4>
+            {website ? <ExternalLink url={website} /> : '--'}
+        </div>
+    );
+
+    const cjBlock = (
+        <div className="agency-overview__data">
+            <h4>Congressional Justification of Budget (CJ)</h4>
+            {congressionalJustification ? <ExternalLink url={congressionalJustification} /> : '--'}
+        </div>
+    );
+
+    // Mobile layout
+    let content = (
+        <>
+            {showAboutData ? aboutBlock : missionBlock}
+            <ReadMore>
+                {showAboutData && missionBlock}
+                {websiteBlock}
+                {cjBlock}
+            </ReadMore>
+        </>
+    );
+
+    // Desktop layout
+    if (windowWidth >= mediumScreen) {
+        content = (
+            <div className="agency-overview__row">
+                <div className="agency-overview__column">
+                    {showAboutData && aboutBlock}
+                    {missionBlock}
+                </div>
+                <div className="agency-overview__column">
+                    {websiteBlock}
+                </div>
+                <div className="agency-overview__column">
+                    {cjBlock}
+                </div>
+            </div>
+        );
+    }
+
+    return isLoading ? <LoadingMessage /> :
```

Comment:
I'm not sure isLoading is working as planned; I see several "--" for a few seconds until the API call finishes.

Reply:
I added the loading message just this morning- are you on the latest commit?

Reply:
wasn't